### PR TITLE
Prepare for 4.0.0

### PR DESCRIPTION
**Pull request by oschwald**
- Make push more robust
- Add .editorconfig
- Update the target frameworks
- Update and simplify dependencies
- Use C# 10
- Replace deprecated option
- Add missing standard exception constructor
- Simplify conditional
- Do not interpolate
- Clarify suppression
- Simplify object creation
- Minor comment fixes
- Use more braces
- Whitespace cleanup
- Style fixes
- Remove deprecated code
- Set release date
- Update CI

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -32,6 +32,11 @@ jobs:
         with:
           dotnet-version: 5.0.*
 
+      - name: Setup .NET .0
```
Comment (ugexe) [inline .github/workflows/test.yml:35]: `6.0`

Comment (oschwald) [reply .github/workflows/test.yml]: Thanks! I force pushed a fix.

File: MaxMind.MinFraud/WebServiceClient.cs [hunk 2]
```diff
@@ -254,7 +255,7 @@ private static async Task Handle4xxStatus(HttpResponseMessage response)
             // set Content for the default response.
             var content = response.Content != null ? await response.Content.ReadAsStringAsync().ConfigureAwait(false) : null;
 
-            if (content == null || content.Length == 0)
+            if (string.IsNullOrEmpty(content))
```
Comment (ugexe) [inline MaxMind.MinFraud/WebServiceClient.cs:258]: The warnings make it look like `netstandard2.0` doesn't treat `IsNullOrEmpty` as a null reference check, is that alright?

Comment (oschwald) [reply MaxMind.MinFraud/WebServiceClient.cs:258]: Yeah, I saw that. It seemed fine to me given that the behavior is still the same. I suppose we could suppress the warning or revert this change, but `netstandard2.0` is the next version to be cut.

**Review by ugexe — APPROVED** [review]

**Pull request merged**